composeApp/src/androidMain/kotlin/org/ooni/probe/MainActivity.kt

```diff
@@ -1,18 +1,48 @@
 package org.ooni.probe
 
+import android.content.Intent
+import android.net.Uri
 import android.os.Bundle
 import androidx.activity.ComponentActivity
 import androidx.activity.compose.setContent
 import androidx.activity.enableEdgeToEdge
+import androidx.navigation.NavHostController
+import co.touchlab.kermit.Logger
 
 class MainActivity : ComponentActivity() {
+    private lateinit var navController: NavHostController
     private val app get() = applicationContext as AndroidApplication
 
     override fun onCreate(savedInstanceState: Bundle?) {
         enableEdgeToEdge()
         super.onCreate(savedInstanceState)
         setContent {
-            App(app.dependencies)
+            navController = App(app.dependencies)
+        }
+    }
+
+    override fun onNewIntent(intent: Intent) {
+        super.onNewIntent(intent)
+        if (Intent.ACTION_VIEW == intent.action) {
+            manageIntent(intent)
+        }
+    }
```

Comment:
I believe `onNewIntent` only works if the app is already running. You need to check for the `getIntent` inside the `onCreate` as well.

Reply:
Yes. And sometimes thats the case.